frontends/ast/simplify.cc

```diff
@@ -300,13 +300,40 @@ static int size_packed_struct(AstNode *snode, int base_offset)
 	bool is_union = (snode->type == AST_UNION);
 	int offset = 0;
 	int packed_width = -1;
+	// embeded struct or union with range?
+	std::vector<AstNode *> ranges;
+	for(auto it = snode->children.begin(); it != snode->children.end(); ) {
```

Comment:
Consider something like this:
```cpp
auto it = std::remove_if(snode->children.begin(), snode->children.end(),
                         [](AstNode* node) { return node->type == AST_RANGE });
std::vector<AstNode*> ranges(it, snode->children.end());
snode->children.erase(it, snode->children.end());
```
It's a bit clearer to me. Not sure if they like this sort of thing upstream.

Reply:
True, it is also more clearer to me. I've also changed error message if there is more then single range.